src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt

```diff
@@ -0,0 +1,26 @@
+package com.hibob.academy.employeeFeedback.service.feedback
+
+import com.hibob.academy.employeeFeedback.dao.FeedbackDao
+import com.hibob.academy.employeeFeedback.model.Feedback
+import com.hibob.academy.employeeFeedback.model.FilterFeedbackRequest
+import com.hibob.academy.employeeFeedback.model.LoggedInUser
+import com.hibob.academy.employeeFeedback.model.StatusResponse
+import org.springframework.stereotype.Service
+
+@Service
+class FeedbackFetcher(
+    private val feedbackDao: FeedbackDao
+) {
+
+    fun getFeedbackById(companyId: Long, feedbackId: Long) : Feedback {
+        return feedbackDao.getFeedbackById(companyId, feedbackId)
+    }
+
+    fun getFeedbacksByCompany(companyId: Long, filters: FilterFeedbackRequest) : List<Feedback> {
+        return feedbackDao.getFeedbacksByFilters(companyId, filters)
+    }
+
+    fun getFeedbackStatusById(companyId: Long, feedbackId: Long) : StatusResponse {
```

Comment:
there's a little redundancy. It's ok just to use `getFeedbackById` and the controller will just return the status. 
I don't remember the dao method - do you use a different mapper just to extract the status?
Anyhow - I think you can just remove this method. Up to you

Reply:
I have a different mapper for the status and a different method for this in the dao.